config/application.rb

```diff
@@ -144,13 +144,19 @@ class Application < Rails::Application
     config.view_component.show_previews = IdentityConfig.store.component_previews_enabled
     if IdentityConfig.store.component_previews_enabled
       require 'lookbook'
+      require 'component_preview_csp'
+      config.middleware.insert_after ActionDispatch::Static, ComponentPreviewCsp
 
       config.view_component.preview_controller = 'ComponentPreviewController'
       config.view_component.preview_paths = [Rails.root.join('spec', 'components', 'previews')]
       config.view_component.default_preview_layout = 'component_preview'
       config.lookbook.auto_refresh = false
       config.lookbook.project_name = "#{APP_NAME} Component Previews"
       config.lookbook.ui_theme = 'blue'
+      if Identity::Hostdata.env == 'dev'
```

Comment:
Tied to my previous comment: We could check the `component_previews_embed_frame_ancestors` here to have a single source of truth.

```suggestion
      if IdentityConfig.store.component_previews_embed_frame_ancestors.present?
```